### PR TITLE
Fix Exit Code in case that one of Reader/Writer fails

### PR DESCRIPTION
**Pull request by rollandf**
`up` will exit with error if the run group finishes with error:
https://github.com/observatorium/up/blob/master/main.go#L262

However, the run group will return the error returned by the first exiting actor:
https://github.com/oklog/run/blob/master/group.go#L28

In general, the writer will finish first.
It can be reproduce in `up` repo by running `make test-integration` with a wrong URL.
Changing endpoint-read will not fail, and changing endpoint-write will fail.

See: https://github.com/observatorium/configuration/issues/211

### CODE REVIEW

**Comment by rollandf** [comment]
@squat @kakkoyun Please take a look.
The CI fails because the `main` function is too long. I will fix it if we are OK with the logic.

File: main.go [hunk 1]
```diff
@@ -259,7 +262,20 @@ func main() {
...
+		level.Info(l).Log("msg", "run group exited with error", "err", err)
+	}
+
+	close(ch)
+
+	fail := false
+	for err := range ch {
+		fail = true
+
+		level.Error(l).Log(err)
```
Comment (kakkoyun) [inline main.go]:
```suggestion
		level.Error(l).Log("err", err)
```

**Comment by rollandf** [comment]
> Overall looks good. Thanks for finding and fixing the issue.
> 
> Rather that pass the channel around, how about using groups, something like:
> 
> ```go
> ctx, cancel := context.WithCancel(context.Background())
> // Assume content or a channel for cancelation properly used.
> var g run.Group
> var w run.Group
> var r run.Group
> g.Add(w.Run, func(err error) {
> 	level.Error(l).Log("msg", "up failed", "err", err)
> 	cancel()
> })
> g.Add(r.Run, func(err error) {
> 	level.Error(l).Log("msg", "up failed", "err", err)
> 	cancel()
> })
> 
> g.Run()
> ```


Won't we get to the same issue again? If the `w` finishes first without error , the `g` won't return any error. And I don't see how to check errors on a group after it has been executed.

> 
> And CI is failing.

Yes, the `main` is too long. I will fix after we agree on logic.

File: main.go (hunk 1 above)
Comment (rollandf) [reply main.go]: Done

**Comment by kakkoyun** [comment]
@rollandf Yes, you're right. Ignore me. I thought we could find a cleaner way with the groups.

**Comment by kakkoyun** [comment]
Let's give others a chance to review as well.

**Comment by rollandf** [comment]
ping

**Comment by brancz** [comment]
Happy with this approach.

**Pull request merged**